### ISSUE TITLE
aws_transfer_user should provide home-directory-mappings option.

### ISSUE THREAD

**Issue by megumitomita** [terraform-providers/terraform-provider-aws#11281]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS CLI https://docs.aws.amazon.com/cli/latest/reference/transfer/create-user.html provides` home-directory-type` and `home-directory-mappings` to control logical mapping between S3 and SFTP. Terraform AWS provider does not support the feature.

### New or Affected Resource(s)

* aws_transfer_user

### Potential Terraform Configuration

```hcl
resource "aws_transfer_user" "sftp_user_dbs" {
  # other options
  # ...
  home_directory_type = "LOGICAL"
  home_directory_mapping = {
    entry = "your-personal-report.pdf"
    target = "/bucket3/customized-reports/${transfer:UserName}.pdf" 
  }
}
```

### References

* https://docs.aws.amazon.com/cli/latest/reference/transfer/create-user.html

**Comment by manpreetsnehra** [comment]
Is this going to be merged?

**Comment by bflad** [comment]
Support for this functionality has been merged and will release with version 3.8.0 of the Terraform AWS Provider, later this week. Thanks to @teraken0509 and @quinyx-tjeerd for the implementation. 👍
